Gems/ScriptCanvas/Code/Editor/Assets/ScriptCanvasFileHandling.cpp

```diff
@@ -289,21 +297,29 @@ namespace ScriptCanvasEditor
             }
         }
 
-        if (auto entity = scriptCanvasData->GetScriptCanvasEntity())
+        auto entity = scriptCanvasData->GetScriptCanvasEntity();
+        if (!entity)
         {
-            AZ_Assert(entity->GetState() == AZ::Entity::State::Constructed, "Entity loaded in bad state");
-            ScriptCanvasFileHandlingCpp::MakeGraphComponentEntityIdsUnique(entity, serializeContext);
-            auto graph = entity->FindComponent<ScriptCanvasEditor::EditorGraph>();
-            graph->MarkOwnership(*scriptCanvasData);
-            entity->Init();
-            entity->Activate();
+            return AZ::Failure(AZStd::string("Loaded script canvas file was missing a necessary Entity."));
         }
-        else
+        else if (entity->GetState() != AZ::Entity::State::Constructed)
```

Comment:
Seems like this is a new check which we didn't consider before?

Reply:
We did, but I asserted on it. In general I remove ScriptCanvas editor tool and build time asserts and replace them with reporting for the caller to send to pop up windows, messages in the nodes themselves, or AP logs as appropriate. I try not to go straight to AZ_Error/Warning/Assert in general, as it makes the functions less useful, more noisy, and if nothing else, spams the console.